### ISSUE TITLE
Missing error for conflicting name of a local method which leads to 'Failed to emit module' compilation error

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#13193]
Test code included at the bottom of this report.

If there is a conflict between local method name and type parameter name of enclosing regular method roslyn fails to detect it. Attempt to compile a solution with such error will lead to 'failed to emit module' error which is extremely unclear and hard to track down.

```cs
class C
{
    public void M<TLocalMethod>()
    {
        void TLocalMethod() // missing error
        {
        }
    }
}
```

**Comment by agocke** [comment]
@TessenR Nice catch!

**Comment by agocke** [comment]
Closed by #14352
